ql/src/java/org/apache/hadoop/hive/ql/optimizer/Optimizer.java

```diff
@@ -122,6 +122,8 @@ public void initialize(HiveConf hiveConf) {
       transformations.add(new SortedDynPartitionOptimizer());
     }
 
+    transformations.add(new SortedDynPartitionTimeGranularityOptimizer());
```

Comment:
No prob. This transformation/optimization is added to the set of transformations that are executed for every query plan.

The transformation itself matches any FileSink in the plan (L94 in SortedDynPartitionTimeGranularityOptimizer), but it bails out fast if the FileSink operator does not use the Druid Storage Handler (L127 in SortedDynPartitionTimeGranularityOptimizer).